### PR TITLE
systemd-sysusers: add policy

### PR DESCRIPTION
**Pull request by bauen1**
Running a system without the unconfined module, this is another service that needs additional privileges (modifying /etc/shadow and friends).
This is blocked by #246 but open for comments.

### CODE REVIEW

File: policy/modules/system/systemd.te [hunk 1]
```diff
@@ -1158,6 +1165,36 @@ seutil_read_file_contexts(systemd_sessions_t)
...
+#########################################
+#
+# Sysusers local policy
+#
+
+allow systemd_sysusers_t self:capability { chown fsetid };
+allow systemd_sysusers_t self:process setfscreate;
+allow systemd_sysusers_t self:unix_dgram_socket { create_socket_perms sendto };
+
+dev_write_kmsg(systemd_sysusers_t)
```
Comment (pebenito) [inline policy/modules/system/systemd.te:1178]: Please see `systemd_log_parse_environment()` which will handle several of the perms here.

File: policy/modules/system/systemd.if [hunk 2]
```diff
@@ -1126,3 +1126,46 @@ interface(`systemd_write_all_user_keys',`
...
+## </param>
+interface(`systemd_domtrans_systemd_sysusers', `
+	gen_require(`
+		type systemd_sysusers_t, systemd_sysusers_exec_t;
+	')
+
+	corecmd_search_bin($1)
+	domtrans_pattern($1, systemd_sysusers_exec_t, systemd_sysusers_t)
+')
+########################################
```
Comment (pebenito) [inline policy/modules/system/systemd.if:1148]: Nit: missing whitespace

File: policy/modules/system/systemd.if [hunk 3]
```diff
@@ -1126,3 +1126,48 @@ interface(`systemd_write_all_user_keys',`
...
+##  Execute systemd-sysusers in the
+##  systemd sysusers domain.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`systemd_domtrans_systemd_sysusers', `
```
Comment (pebenito) [inline policy/modules/system/systemd.if:1141]: I think we should drop the extra "systemd" in the interface names here and below.

File: policy/modules/system/systemd.te [hunk 4]
```diff
@@ -1158,6 +1165,34 @@ seutil_read_file_contexts(systemd_sessions_t)
...
+#########################################
+#
+# Sysusers local policy
+#
+
+allow systemd_sysusers_t self:capability { chown fsetid };
+allow systemd_sysusers_t self:process setfscreate;
+allow systemd_sysusers_t self:unix_dgram_socket sendto;
+
+dev_write_kmsg(systemd_sysusers_t)
```
Comment (pebenito) [inline policy/modules/system/systemd.te:1178]: This should be handled by `systemd_log_parse_environment()`

Comment (bauen1) [reply policy/modules/system/systemd.te:1178]: I may have messed up git, I'll fix this.

**Pull request merged**